crates/evm/src/machine.cairo

```
@@ -129,6 +129,23 @@ impl MachineCurrentContextImpl of MachineCurrentContextTrait {
         call_ctx
     }
 
+    /// Returns from the sub context by setting the current context
+    /// to the parent context.
+    #[inline(always)]
+    fn return_to_parent_ctx(ref self: Machine) {
+        let mut current_ctx = self.current_ctx.unbox();
+        let maybe_parent_ctx = current_ctx.parent_ctx;
+        match match_nullable(maybe_parent_ctx) {
+            FromNullableResult::Null => current_ctx = Default::default(),
```

Why do we have to return Default::default() from this match arm?

---

if I don't do anything with it I get `Variable was previously moved.` Do you want to keep the current ctx? Or return an error?

---

I think we can return an error and introduce a result based return signature